src/app/Http/Controllers/Admin/UniquePageCrudController.php

```diff
@@ -0,0 +1,190 @@
+
+        return parent::edit($entry->id);
+    }
+
+    public function update($slug, $id)
+    {
+        $this->setRoute($slug);
+
+        return parent::updateCrud();
+    }
+
+    public function setRoute($slug)
+    {
+        $this->crud->setRoute(config('backpack.base.route_prefix').'/unique/'.$slug);
+    }
+
+    /**
+     * Populate the update form with basic fields, that all pages need.
+     *
+     * @param Model $page the page entity
+     */
+    public function addDefaultPageFields($page)
+    {
+        $this->crud->addField([
+            'name' => 'template',
+            'type' => 'hidden',
+        ]);
+        $this->crud->addField([
+            'name' => 'name',
+            'type' => 'hidden',
+        ]);
+        $this->crud->addField([
+            'name' => 'title',
+            'type' => 'hidden',
+        ]);
+        $this->crud->addField([
+            'name' => 'slug',
+            'type' => 'hidden',
+        ]);
+
+        $this->crud->addField([
+            'name' => 'open',
+            'type' => 'custom_html',
+            'value' => $this->buttons($page),
+        ]);
+    }
+
+    public function buttons($page)
+    {
+        $openButton = $page->getOpenButton();
+        $revisionsButton = view('crud::buttons.revisions', ['crud' => $this->crud, 'entry' => $page]);
+
+        return $openButton.' '.$revisionsButton;
+    }
+
+    public function createMissingPage($slug)
+    {
+        $pages = collect($this->getUniquePages());
+
+        $slugs = $pages->mapWithKeys(function ($page) {
+            return [str_slug($page->name) => $page->name];
+        });
+
+        if (! $page = $slugs->pull($slug)) {
+            abort(404);
+        }
+
+        $model = $this->crud->model;
+
+        return $model::create([
+            'template' => $page,
+            'name' => camel_case($page),
+            'title' => camel_case($page),
+            'slug' => $slug,
+        ]);
+    }
+
+    public function uniqueRevisions($slug, $id)
+    {
+        $model = $this->crud->model;
+        $entry = $model::findBySlugOrFail($slug);
+
+        $this->uniqueSetup($entry);
+
+        return parent::listRevisions($entry->id);
+    }
+
+    public function restoreUniqueRevision($slug, $id)
+    {
+        $model = $this->crud->model;
+        $entry = $model::findBySlugOrFail($slug);
+
+        $this->uniqueSetup($entry);
+
+        return parent::restoreRevision($id);
+    }
+
+    protected function uniqueSetup($entry)
+    {
+        $this->setRoute($entry->slug);
+
+        $this->addDefaultPageFields($entry);
+        $this->crud->setEntityNameStrings($this->crud->makeLabel($entry->template), '');
+
+        $this->{$entry->template}();
+    }
+
+    /*
+    |--------------------------------------------------------------------------
+    | SaveActions overrides
+    |--------------------------------------------------------------------------
+    */
+
+    /**
+     * Overrides trait version to remove.
```

Comment:
...and has no @return in the PHP doc.

This might cause some grief if someone has overridden these themselves.